stripe/src/main/java/com/stripe/android/StripeApiHandler.java

```diff
@@ -133,6 +135,156 @@ static Source createSource(
                 null);
     }
 
+
+    @Nullable
+    static PaymentIntent confirmPaymentIntent(
+            @Nullable StripeNetworkUtils.UidProvider uidProvider,
+            @NonNull Context context,
+            @NonNull PaymentIntentParams paymentIntentParams,
+            @NonNull String publishableKey,
+            @Nullable String stripeAccount,
+            @Nullable LoggingResponseListener loggingResponseListener)
+            throws AuthenticationException,
+            InvalidRequestException,
+            APIConnectionException,
+            APIException {
+        Map<String, Object> paramMap = paymentIntentParams.toParamMap();
+        StripeNetworkUtils.addUidParamsToPaymentIntent(uidProvider, context, paramMap);
+        RequestOptions options = RequestOptions.builder(
+                publishableKey,
+                stripeAccount,
+                RequestOptions.TYPE_QUERY).build();
+
+        try {
+            String apiKey = options.getPublishableApiKey();
+            if (StripeTextUtils.isBlank(apiKey)) {
+                return null;
+            }
+
+            setTelemetryData(context, loggingResponseListener);
+            SourceParams sourceParams = paymentIntentParams.getSourceParams();
+            String sourceType = sourceParams != null ? sourceParams.getType() : null;
+            Map<String, Object> loggingParams = LoggingUtils.getPaymentIntentConfirmationParams(
+                    context,
+                    null,
```

Comment:
Should this be non-null? Or can it be removed from the method signature?

Reply:
I think they would be come non-null if we were to ever attach it to a UI element.